Enhanced: Support for IBM Domino POP3 task

cbrandlehner opened the pull request:
Before submitting your PR, please review the following checklist:

- [ ] **CHOOSE CORRECT BRANCH**: if filing a bugfix/enhancement
      against 0.9.x series, choose `master` branch
- [ ] **CONSIDER adding a unit test** if your PR resolves an issue
- [ ] **LIST ISSUES** this PR resolves
- [ ] **MAKE SURE** this PR doesn't break existing tests
- [ ] **KEEP PR small** so it could be easily reviewed.
- [ ] **AVOID** making unnecessary stylistic changes in unrelated code
- [ ] **ACCOMPANY** each new `failregex` for filter `X` with sample log lines
      within `fail2ban/tests/files/logs/X` file

cbrandlehner commented:
Hmm, I see no real reason why travis failed based on the changes I submitted.

sebres commented on config/filter.d/domino-pop3.conf:
```diff
@@ -0,0 +1,38 @@
+# 03.10.2016 10:50:01   POP3 Server: Authentication failure for httpd, connecting host 58.1.245.164: Password not found in the Name and Address Book entry or password did not verify
+# 03.10.2016 10:51:19   POP3 Server: Authentication failure for noreply@github.com, connecting host 80.122.241.130: Password not found in the Name and Address Book entry or password did not verify
+
+failregex = .* POP3 Server: Authentication failure for .* connecting host <HOST>: Password not found
```
The CI-error says:
> AssertionError: Following regexp of "domino-pop3" contains greedy catch-all before `<HOST>`, that is not hard-anchored at end or has not precise sub expression after `<HOST>`:
> `.* POP3 Server: Authentication failure for .* connecting host <HOST>: Password not found`

I've extended our test-cases to cover exactly such situations.
Although your expression can work, but is vulnerable (see abovementioned error):
- multiple catch-all before `<HOST>` token 
- neither it is anchored nor it is precise enough;

cbrandlehner commented:
I hope my pull request improved.

cbrandlehner commented:
please advise what to change. I was hoping my pull request improved the second time.

sebres commented on config/filter.d/domino-pop3.conf:
```diff
@@ -0,0 +1,38 @@
+# 03.10.2016 10:50:01   POP3 Server: Authentication failure for httpd, connecting host 58.1.245.164: Password not found in the Name and Address Book entry or password did not verify
+# 03.10.2016 10:51:19   POP3 Server: Authentication failure for noreply@github.com, connecting host 80.122.241.130: Password not found in the Name and Address Book entry or password did not verify
+
+failregex = ^(\[[0-9]{5}:[0-9]{5}-[0-9]{10}])?.*   POP3 Server: Authentication failure for .* connecting host <HOST>: Password not found in the Name and Address Book entry or password did not verify\s*$
```
```diff
-failregex = ^(\[[0-9]{5}:[0-9]{5}-[0-9]{10}])?.*   POP3 Server: Authentication failure for .* connecting host <HOST>: Password not found in the Name and Address Book entry or password did not verify\s*$
+failregex = ^(?:\[[^]]*\])?\s*POP3 Server: Authentication failure for (?:[^,]+|.*?), connecting host <HOST>: Password not found in the Name and Address Book entry or password did not verify\s*$
```

cbrandlehner commented:
@sebres  I updated my files as requested and updated the pull request. However, I have no idea how to solve the two remaining failing checks codecov/project and coverage/coveralls.

sebres commented:
Don't worry, you don't have changed the code, so this are just sporadic (run-time related) coverage decreases.

sebres commented:
BTW.
Why you discarded my suggested `(?:[^,]+|.*?)` (replaced with greedy catch-all `.*`)?

Suggested expression was safer (in order to avoid slowing down on long stings by injection on username), and at least faster in normal case, for the regular user names (that normally don't contain comma) and avoid to enter catch-all branch at all (which would be rather an exception case).

cbrandlehner commented:
It did not work for me in a test a did but I will give it a new try the next days as it was most likely my fault.
Thanks for letting me know.

cbrandlehner commented:
I did not update this PR as it is already approved. Any plans to merge it?
